src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/AddExpressionLanguageProvidersPass.php

```diff
@@ -36,10 +36,10 @@ public function process(ContainerBuilder $container)
         }
 
         // security
-        if ($container->has('security.access.expression_voter')) {
-            $definition = $container->findDefinition('security.access.expression_voter');
+        if ($container->has('security.expression_language')) {
```

Comment:
why changing the registration logic ?

Reply:
hmm, registering providers on the right service rather than relying on delegation still makes sense though, to ensure that the `security.expression_language` service is usable even if it is instantiated earlier than the ExpressionVoter due to injecting it elsewhere and due to voters being lazy-loaded).
We should probably even deprecated `\Symfony\Component\Security\Core\Authorization\Voter\ExpressionVoter::addExpressionLanguageProvider` in master as it only delegates the call and the ExpressionLanguage instance is always injected from the outside.

Reply:
I think adding it to `security.expression_language` directly makes more sense? Not even sure why `ExpressionVoter::addExpressionLanguageProvider` exists. Maybe I'm missing something?

If someone injects `security.expression_language` and uses it without the `ExpressionVoter` service ever being instantiated then the custom providers would be missing?

But if it makes merging etc harder I can also revert it.

Reply:
@stof exactly :blush: 

:+1: for deprecating it

Reply:
is this change required for 2.7? if not, better revert, isn't it, and do it on master only?

Reply:
No its not required :+1: will revert and apply to master together with the deprecation then

Reply:
@nicolas-grekas removed :wink: